forum/api/threads.py

```diff
@@ -262,6 +262,11 @@ def update_thread(
             )
     backend.update_thread(thread_id, **update_thread_data)
     thread = backend.get_thread(thread_id)
+    if thread is None:
+        log.error(
+            "Forumv2RequestError for update thread request - retrieving updated thread to send in response."
```

Comment:
```suggestion
            "ForumV2RequestError for update thread request - retrieving updated thread to send in response."
```
This is the only part I wasn't sure about. It says "retrieving updated thread" but I don't see it doing that here.

Reply:
The general flow in this method (pseudocode) is:

- thread: Thread = get_thread_or_raise_error()
- update_thread()
- thread: Thread|None = get_thread_or_none(id)



So on line 265, thread should be the thread after the updates on line 263. However, nothing in the type system guarantees the thread wasn't deleted between line 263 and line 264. So the check is to keep the types happy, although there's a very very small chance of that happening.